src/Bundle/CoreBundle/Form/StateType.php

```diff
@@ -0,0 +1,293 @@
+                ->atPath('[state][transition]')
+                ->addViolation();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        if (isset($view->vars['label']))
+        {
+            $view->vars['widget_label'] = $view->vars['label'];
+            $view->vars['label'] = false;
+        }
+
+        if ($view->vars['value']['state']) {
+            $view->vars['current_state'] = $this->translator->trans('state.field.current_state');
+            $view->vars['current_state'] .= $this->settings['places'][$view->vars['value']['state']]['label'];
+        }
+
+        parent::buildView($view, $form, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $resolver->setDefaults([
+            'label_prefix' => 'state',
+            'settings' => [],
+            'compound' => true
+        ]);
+
+    }
+
+    /**
+     * Check if can transist from current state
+     *
+     * @param string $current_state
+     * @param string $transition_to
+     *
+     * @return bool
+     */
+    public function canTransist(string $current_state, string $transition_to) : bool
+    {
+        // transitions not set
+        if (!isset($this->settings['transitions'][$transition_to]))
+        {
+            return FALSE;
+        }
+
+        $transition = $this->settings['transitions'][$transition_to];
+
+        if (in_array($current_state, $transition['from']))
+        {
+            return TRUE;
+        }
+
+        return FALSE;
+    }
+
+     /**
+     * @param array $transitions
+     *
+     * @return array
+     */
+    private function getChoices(array $transitions) : array
+    {
+        $choices = [];
+
+        foreach ($transitions as $transition_key => $transition) {
+            $choices[$transition['label']] = $transition_key;
+        }
+
+        return $choices;
+    }
+
+    /**
+     * @param string $transition_key
+     *
+     * @return string
+     */
+    private function getPlaceCategory(string $transition_key) : string
+    {
+        $place = $this->settings['transitions'][$transition_key]['to'];
+
+        if (isset($this->settings['places'][$place]['category'])) {
+            return $this->settings['places'][$place]['category'];
+        }
+
+        return "";
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($value)
+    {
+        $new_values = [];
+        
+        $new_values['state'] = $value;
+        
+        // transition can be alway null
+        $new_values['transition'] = null;
+
+        // if no state set (new content), take initial place
+        if (empty($value))
+        {
+            $new_values['state'] = $this->settings['initial_place'];
+        }
+
+        return $new_values;

```

Comment:
Why do you return an array with an empty transition value here and not only the state (as a string instead of an error). I think we just need to save the string to the database and not the array, or am i wrong?